extensions/amp-base-carousel/1.0/amp-base-carousel.js

```diff
@@ -84,6 +84,24 @@ class AmpBaseCarousel extends PreactBaseElement {
       this.api().goToSlide(slide);
     }
   }
+
+  /** @override */
+  updatePropsForRendering(props) {
+    if (props['arrowPrev']) {
+      const Comp = props['arrowPrev'];
+      props['arrowPrevAs'] = (props) => {
+        Comp.props = {...Comp.props, ...props};
```

Comment:
Can you explain what this is trying to do? What is `Comp` expected to be before this?

Reply:
+1, I'm not sure what changing `Comp.props` does.

Reply:
`Comp` here is the `Slot` made from the `selector`-based code:
https://github.com/ampproject/amphtml/blob/e03dba1b1fe7ff1d4dfeb59ba257ab4d0d93341f/src/preact/base-element.js#L1096-L1102

So `Comp.props` is the props given  by creating the Slot in that operation. The parameter `props` are the ones given by BaseCarousel when using `arrowPrevAs` as the `Comp` here:
https://github.com/ampproject/amphtml/blob/625e06dd9cc9a607270cdd1fa9f1669fe741a796/extensions/amp-base-carousel/1.0/arrow.js#L41-L59

Reply:
An important example for adding onto the props is that without this operation (if we just did `props['arrowPrevAs'] = () => props['arrowPrev']`) is that we would lose the `onClick` handler given by the carousel `Arrow` class to transition the slide. In AMP pages, without it, the publisher would have to add `on=tap:carousel.next` to their custom arrows, which is what we did with the custom lightbox example (`on=tap:lightbox.close`) but is not ideal as it's not intuitive and reasonable to expect should be unnecessary.

Reply:
Ok, so `Comp` here isn't actually a Component, it's a JSX node, and we should lower case it at least. More likely, we should change this to be:

```js
const {arrowPrev} = props;
props['arrowPrevAs'] = (props) => {
  Preact.cloneElement(arrowPrev, props);
};
```

Reply:
Gotcha, thank you. Fixed!